src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -515,12 +515,19 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
 	}
 	
 	private func handleCoronaTestRegistrationInformation(_ testRegistrationInformation: CoronaTestRegistrationInformation, isLoading: @escaping (Bool) -> Void) {
+		var isConsentGiven: Bool
+		switch testRegistrationInformation {
+		case .teleTAN:
+			isConsentGiven = false
+		case .pcr, .antigen:
+			isConsentGiven = true
+		}
```

Comment:
```suggestion
	private func handleCoronaTestRegistrationInformation(_ testRegistrationInformation: CoronaTestRegistrationInformation, submissionConsentGiven: Bool, isLoading: @escaping (Bool) -> Void) {
```

I think I would prefer to set it on the calling side, because it feels like it should depend on the flow that one is coming from instead of the registration type. That's at least what I had in mind when we were talking earlier. But this solution is not wrong. I checked it out and tried it, worked correctly for me. 👍

If you want to leave it like it is, I'll approve, if you want to change it the way I suggested, I can check it out again :)

Reply:
i will change it